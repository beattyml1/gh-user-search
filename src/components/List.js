import * as React from 'react';
import './List.css';
export const List = ({items, getItemProps, itemComponent}) => {
    let ItemComponent = itemComponent;
    getItemProps = getItemProps || (x => x);
    return (
        <ul className={"list"} role={"list"}>
            {items.map(item =>
                <div className={"list-item"} role={"listitem"}>
                    <ItemComponent {...getItemProps(item)} />
                </div>
            )}
        </ul>
    );
}