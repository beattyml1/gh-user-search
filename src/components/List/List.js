import * as React from 'react';
import './List.css';
export const List = ({items, getItemProps, itemComponent, getKey}) => {
    let ItemComponent = itemComponent;
    getItemProps = getItemProps || (x => x);
    return (
        <ul className={"list"}>
            {items.map(item =>
                <div className={"list-item"} role={"listitem"} key={getKey ? getKey(item) : item}>
                    <ItemComponent {...getItemProps(item)} />
                </div>
            )}
        </ul>
    );
}