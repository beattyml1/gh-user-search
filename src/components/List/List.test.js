import * as React from 'react';
import ReactDOM from 'react-dom';
import {List} from './List';
import renderer from 'react-test-renderer';

it('creates expected mark up for 2 results', () => {
    const tree = renderer.create(
        <List items={ [{a:'a1', b: 'b1'}, {a:'a2', b: 'b2'}] } itemComponent={i=> <div>{i.a}</div>} getKey={i=>i.a}/>
    ).toJSON();
    expect(tree).toMatchSnapshot()
});


it('creates expected mark up for 0 results', () => {
    const tree = renderer.create(
        <List items={ [] } itemComponent={i=> <div>{i.a}</div>}/>
    ).toJSON();
    expect(tree).toMatchSnapshot()
});