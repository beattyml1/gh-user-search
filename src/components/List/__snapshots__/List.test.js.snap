// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creates expected mark up for 0 results 1`] = `
<ul
  className="list"
/>
`;

exports[`creates expected mark up for 2 results 1`] = `
<ul
  className="list"
>
  <div
    className="list-item"
    role="listitem"
  >
    <div>
      a1
    </div>
  </div>
  <div
    className="list-item"
    role="listitem"
  >
    <div>
      a2
    </div>
  </div>
</ul>
`;
